### PR TITLE
Add function to extend listeners

### PR DESCRIPTION
**Pull request by sapsh-msft**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK/Framework/Managers/ConfigurationManager.ps1 [hunk 1]
```diff
@@ -57,22 +57,93 @@ class ConfigurationManager
...
-                $extensionScriptCode = [ConfigurationManager]::LoadServerFileRaw($extensionSVTClassFileName);
-                if(-not [string]::IsNullOrWhiteSpace($extensionScriptCode))
-                {
-                    $extensionFilePath = "$([Constants]::AzSKExtensionsFolderPath)\$extensionSVTClassFileName";
-                    Out-File -InputObject $extensionScriptCode -Force -FilePath $extensionFilePath -Encoding utf8;                                                                            
-                }
+				$extensionFilePath = [ConfigurationManager]::DownloadExtFile($extensionSVTClassFileName)
 			}
 			catch {
 				Write-host $_;
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Managers/ConfigurationManager.ps1]: Please remove the write-host...I had it for debug. forgot to remove

File: src/AzSK/Framework/Managers/ConfigurationManager.ps1 [hunk 2]
```diff
@@ -57,22 +57,93 @@ class ConfigurationManager
...
+							$extensionFilePath = [ConfigurationManager]::DownloadExtFile($listenerFileName)
+							
+							# file has to be loaded here due to scope constraint
+							. $extensionFilePath
+							
+							$listenerClassName = $listenerFileName.trimend(".ext.ps1") + "Ext"
+							Invoke-Expression "[$listenerClassName]::GetInstance().RegisterEvents();"
+						}
+						catch {
+							Write-host $_;
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Managers/ConfigurationManager.ps1]: Please remove the write-host...I had it for debug. forgot to remove

File: src/AzSK/Framework/Managers/ConfigurationManager.ps1 [hunk 3]
```diff
@@ -57,22 +57,93 @@ class ConfigurationManager
...
+							
+							# file has to be loaded here due to scope constraint
+							. $extensionFilePath
+
+							$listenerClassName = $listenerFileName.trimend(".ext.ps1") + "Ext"
+							Invoke-Expression "[$listenerClassName]::GetInstance().UnregisterEvents();"
+						}
+						catch {
+							Write-host $_;
+						}
```
Comment (sbyna-msft) [inline src/AzSK/Framework/Managers/ConfigurationManager.ps1]: Please remove the write-host...I had it for debug. forgot to remove

Comment (sapsh-msft) [reply src/AzSK/Framework/Managers/ConfigurationManager.ps1]: Updated catch block to [EventBase]::PublishGenericException($_);

File: src/AzSK/Framework/Managers/ConfigurationManager.ps1 (hunk 2 above)
Comment (sapsh-msft) [reply src/AzSK/Framework/Managers/ConfigurationManager.ps1]: Updated catch block to [EventBase]::PublishGenericException($_);

**Pull request merged**